runtime/src/fips.rs

```diff
@@ -80,13 +59,99 @@ impl FipsVersionCmd {
     }
 }
 
-pub struct FipsSelfTestCmd;
-impl FipsSelfTestCmd {
-    pub(crate) fn execute(env: &mut Drivers) -> CaliptraResult<MailboxResp> {
+#[cfg(feature = "fips_self_test")]
+pub mod fips_self_test_cmd {
+    use super::*;
+    use caliptra_common::{verifier::FirmwareImageVerificationEnv, FMC_ORG, RUNTIME_ORG};
+    use caliptra_drivers::ResetReason;
+    use caliptra_image_verify::ImageVerifier;
+    use zerocopy::AsBytes;
+    use crate::RtBootStatus::{RtFipSelfTestComplete, RtFipSelfTestStarted};
+
+    pub enum SelfTestStatus {
+        Idle,
+        InProgress(fn(&mut Drivers) -> CaliptraResult<()>),
+        Done,
+    }
+
+    fn copy_and_verify_image(env: &mut Drivers) -> CaliptraResult<()> {
+        cprintln!("write dummy cmd");
+        env.mbox.write_cmd(0)?;
+        cprintln!("set dlen");
+        env.mbox
+            .set_dlen(env.manifest.size + env.manifest.fmc.size + env.manifest.runtime.size);
+
+        cprintln!("copy manifest");
+        env.mbox.copy_bytes_to_mbox(env.manifest.as_bytes())?;
+
+        cprintln!("copy fmc");
+        let fmc = unsafe { crate::create_slice(FMC_ORG, env.manifest.fmc.size as usize) };
+        env.mbox.copy_bytes_to_mbox(fmc.as_bytes())?;
+        cprintln!("copy rt");
+        let rt = unsafe { crate::create_slice(RUNTIME_ORG, env.manifest.runtime.size as usize) };
+        env.mbox.copy_bytes_to_mbox(rt.as_bytes())?;
```

Comment:
We could maybe consider putting the rt and fmc regions in the MemoryRegions driver. This would allow rust to enforce accesses better.

Reply:
I believe the memory regions will be replaced by the persistent data accessor soon. Would it be ok if we defer it to a later PR?

Reply:
Ya, that's ok with me